R/install.snowball.R

```diff
@@ -70,38 +72,79 @@ install.snowball <- function(pkg, date, include.suggests, force.install = FALSE,
-      if (!now.installed) {
-        install.source(pkg_vrs.k, lib.k, date, force.download = TRUE, quiet = quiet.install)
+        # If archive version
+        else {
+          url2 <- paste0("https://cran.r-project.org/src/contrib/Archive/", pkg.k, "/", pkg_vrs.k, ".tar.gz")
+          install.packages(url2, repos = NULL, lib = lib.k, type = "source", dependencies = FALSE, quiet = quiet.install)
+        }
+
+        # 2) Try MRAN
+        # Attempt 3 overall, MRAN around selected date, current tarballs path
+        if (!is.pkg_vrs.installed(pkg.k, vrs.k)) # if still not installed
+          {
+            # Set date for mran, midpoint of available dates around requested date  +-10 days,
+            date <- as.Date(date) # Ensure formated as date
+            max.date <- min(Sys.Date() - 2, date + 10) # use as highest possible mran date, 2 days ago
+            mran.date <- get.available.mran.date(date - 10, max.date) # Function get.available.mran.date() in utils.R
+
+            # Try path for current packages
+            url3 <- paste0("https://cran.microsoft.com/snapshot/", mran.date, "/src/contrib/", pkg_vrs.k, ".tar.gz")
+            install.packages(url3, repos = NULL, lib = lib.k, type = "source", dependencies = FALSE, quiet = quiet.install)
+          }
+        # Attempt 4 overall, MRAN around selected date, Archive tarballs path
+        if (!is.pkg_vrs.installed(pkg.k, vrs.k)) # if still not installed
+          {
+            url4 <- paste0("https://cran.microsoft.com/snapshot/", mran.date, "/src/contrib/Archive/", pkg.k, "/", pkg_vrs.k, ".tar.gz")
+            install.packages(url4, repos = NULL, lib = lib.k, type = "source", dependencies = FALSE, quiet = quiet.install)
+          }
+
+        # Attempt 5, MRAN around Publish date, current path
+        if (!is.pkg_vrs.installed(pkg.k, vrs.k)) # if still not installed
+          {
+            # Set date for mran, midpoint of available dates around requested date  +-10 days,
+            toc.pkg <- toc(pkg.k)
+            date.Published <- toc.pkg$Published[toc.pkg$Version == vrs.k] + 2
+            max.date.published <- min(Sys.Date() - 2, date.Published + 20) # Use as highest possible mran published date, 2 days ago
+            mran.published.date <- get.available.mran.date(date.Published, max.date.published) # Function get.available.mran.date() in utils.R
+            # Treat as current
+            url5 <- paste0("https://cran.microsoft.com/snapshot/", mran.published.date, "/src/contrib/", pkg_vrs.k, ".tar.gz")
+            install.packages(url5, repos = NULL, lib = lib.k, type = "source", dependencies = FALSE, quiet = quiet.install)
+          }
+
+        # Attempt 6, MRAN around Publish date, Archive path
+        if (!is.pkg_vrs.installed(pkg.k, vrs.k)) # if still not installed
+          {
+            # If fails try archive
+            url6 <- paste0("https://cran.microsoft.com/snapshot/", mran.published.date, "/src/contrib/", pkg_vrs.k, ".tar.gz")
```

Comment:
@urisohn, `url5` and `url6` are the same. I'm assuming this is a typo?
```suggestion
            url6 <- paste0("https://cran.microsoft.com/snapshot/", mran.published.date, "/src/contrib/Archive/", pkg_vrs.k, ".tar.gz")
```

Reply:
yes, good catch